src/AWSCredentials.jl

```diff
@@ -450,23 +503,41 @@ function aws_get_role(role::AbstractString, ini::Inifile)
     end
 
     credentials === nothing && return nothing
-
     config = AWSConfig(:creds=>credentials, :region=>aws_get_region(source_profile, ini))
 
     role = Services.sts(
         config,
         "AssumeRole",
         RoleArn=role_arn,
-        RoleSessionName=replace(role, r"[^\w+=,.@-]" => s"-"),
+        RoleSessionName=replace(role, r"[^\w+=,.@-]" => s"-")
     )
+
     role_creds = role["Credentials"]
 
-    AWSCredentials(role_creds["AccessKeyId"],
+    return AWSCredentials(
+        role_creds["AccessKeyId"],
         role_creds["SecretAccessKey"],
         role_creds["SessionToken"];
-        expiry = unix2datetime(role_creds["Expiration"]))
+        expiry=unix2datetime(role_creds["Expiration"])
+    )
 end
 
-#==============================================================================#
-# End of file.
-#==============================================================================#
+
+"""
+    aws_get_role_details(profile::AbstractString, ini::Inifile)
+
+# Arguments
+- `profile::AbstractString`: Specific profile to get role details about
+- `ini::Inifile`: Inifile to look into to find the role details
+
+Return a tuple of `profile` details and the `role arn`.
+"""
+function aws_get_role_details(profile::AbstractString, ini::Inifile)
+    role_arn = get(ini, profile, "role_arn")
+    source_profile = get(ini, profile, "source_profile")
+    profile = "profile $profile"
+    role_arn = get(ini, profile, "role_arn", role_arn)
+    source_profile = get(ini, profile, "source_profile", source_profile)
```

Comment:
This `"$profile"` then `"profile $profile"` logic keeps coming up. Consider making a function to abstract this logic?